fhir-client/src/main/java/com/ibm/fhir/client/impl/FHIRClientImpl.java

```diff
@@ -219,6 +222,39 @@ public FHIRResponse conditionalCreate(JsonObject resource, FHIRParameters parame
 
         return result;
     }
+    
+    /**
+     * This function will add a "Prefer" request header to the specified array of headers.
+     * @param headers a possibly null array of FHIRRequestHeader objects
+     * @param  returnPref a value representing the HTTP return preference to be used for the request
+     * @return a new array of FHIRRequestHeader objects containing the additional "Prefer" request header
+     */
+    private FHIRRequestHeader[] addHttpPreferHeader(FHIRRequestHeader[] headers, HTTPReturnPreference returnPref) {
+        if (headers != null ) {
+            for (FHIRRequestHeader fhirRequestHeader : headers) {
+                if ("Prefer".equals(fhirRequestHeader.getName())) {
+                    // User has provided an explicit Prefer header, so don't overwrite that
+                    return headers;
+                }
+            }
+        }
+        
+        FHIRRequestHeader preferHeader = new FHIRRequestHeader("Prefer", "return=" + returnPref.value());
+        
+        // Create a new array that has room for the new "Prefer" header.
+        int headersSize = (headers != null ? headers.length : 0);
+        FHIRRequestHeader[] result = new FHIRRequestHeader[headersSize + 1];
+        if (headers != null) {
+            for (int i = 0; i < headers.length; i++) {
+                result[i] = headers[i];
+            }
+        }
+        
+        // Add the new header at the end.
+        result[result.length - 1] = preferHeader;
```

Comment:
What happens if result.length = 0?

Reply:
its initialized just above with `FHIRRequestHeader[] result = new FHIRRequestHeader[headersSize + 1];` ...I don't think length can be 0 here